pkg/controller/queuejobresources/genericresource/genericresource.go

```diff
@@ -311,8 +311,8 @@ func (gr *GenericResources) SyncQueueJob(aw *arbv1.AppWrapper, awr *arbv1.AppWra
 	// Check to see if object already exists in etcd, if not, create the object.
 	if inEtcd == nil || len(inEtcd.Items) < 1 {
 		newName := name
-		if len(newName) > 63 {
-			newName = newName[:63]
+		if len(newName + namespace) > 63 {
+			newName = newName[:len(newName) - (len(newName) + len(namespace) - 63)]
```

Comment:
at quick glance, I'm not 100% sure about the correctness of these lines. 
but actually, even the original code:
```
	if len(newName) > 63 {
			newName = newName[:63]
```
makes me doubtful. Do we really want to truncate the name at 64 chars? 
wouldn't it lead to unexpected clashes when using similar-but-ending-differently long names? 

Raising an error could be saner (and thus making the AppWrapper impossible to process), it's a common K8s thing to abort on long names.

Reply:
I have tried in the past to concatenate the namespace and name of a job and observed that this would indeed run long for many of our jobs. The use of the hyphen is also problematic because it is ambiguous. We should have separate labels for the name and namespace.

Reply:
Thank you @kpouget, you raised a very good point. For what I could read, it seems that it is done to meet with DNS standards [RFC 1035/1123](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#rfc-1035-label-names:~:text=constraints%20for%20resources.-,DNS%20Subdomain%20Names,end%20with%20an%20alphanumeric%20character,-Path%20Segment%20Names). Since Kubernetes namespaces and names are often used to form a DNS subdomain, K8s enforces it here.

After looking at this code in particular again, I can confirm that we can keep the original code here as the namespace is used in the creation of the object anyways.

Now, to deal with very long names, we could raise an error and stop the operation as suggested. But I was thinking what if we handle the error this way? Here are some ideas:
- Hashing the name for uniqueness?
- Or, Truncate to 60 characters and append i.e., 3 random numbers?

What do you think?

Reply:
Thanks @tardieu, having separate labels for the name and namespace does look like the best option instead of using a hyphen.

Reply:
> Now, to deal with very long names, we could raise an error and stop the operation as suggested. But I was thinking what if we handle the error this way?
do you know how this DNS name is used? is it something that the AppWrapper owner will ever want to use? 
if it's purely internal, then using a hash, or truncating the name + random chars seems to be a good idea